### PR TITLE
use constants from chartutil

### PR DESCRIPTION
**Pull request by bacongobbler**
This way, there's less overall code duplication.

### CODE REVIEW

File: pkg/chartutil/save.go [hunk 1]
```diff
@@ -103,14 +102,14 @@ func Save(c *chart.Chart, outDir string) (string, error) {
...
 
 	if c.Metadata == nil {
-		return "", errors.New("no Chart.yaml data")
+		return "", ErrChartMetadataMissing
 	}
 
 	cfile := c.Metadata
 	if cfile.Name == "" {
-		return "", errors.New("no chart name specified (Chart.yaml)")
+		return "", ErrChartNameEmpty
```
Comment (bacongobbler) [inline pkg/chartutil/save.go]: Note that this error message has been changed to be more consistent with other uses of this error. Please let me know if I should revert this.

File: pkg/chartutil/save.go [hunk 2]
```diff
@@ -103,14 +102,14 @@ func Save(c *chart.Chart, outDir string) (string, error) {
...
 	} else if cfile.Version == "" {
-		return "", errors.New("no chart version specified (Chart.yaml)")
+		return "", ErrChartVersionEmpty
```
Comment (bacongobbler) [inline pkg/chartutil/save.go]: Note that this error message has been changed to be more consistent with other uses of this error. Please let me know if I should revert this.

File: pkg/chartutil/errors.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+*/
+
+package chartutil
+
+import (
+	"fmt"
+)
+
+var (
+	// ErrChartMetadataMissing is raised when a Chart.yaml is missing from a chart.
```
Comment (jascott1) [inline pkg/chartutil/errors.go]: Would it be valuable to use full-on custom error types here? The advantage being it enables the caller to do a type assertion but Im not sure if thats applicable here.

Comment (bacongobbler) [inline pkg/chartutil/errors.go]: Isn't that's what is happening here? e.g. in load_test.go it is doing type assertion that the error is of type ErrChartMetadataMissing down below. :) it's just a matter of downstream packages of pkg/chartutil to use these error types moving forward.

**Comment by bacongobbler** [comment]
rebased, this should be good to go if someone wants to give a quick check!

**Pull request closed without merging**